src/hotspot/share/opto/vectorization.hpp

```diff
@@ -445,6 +451,39 @@ class VLoopTypes : public StackObj {
   const Type* container_type(Node* n) const;
 };
 
+// Submodule of VLoopAnalyzer.
+// We compute and cache the VPointer for every load and store.
+class VLoopPointers : public StackObj {
+private:
+  Arena*                   _arena;
```

Comment:
Will the pointer ever change? Could potentially change this to a reference.

Reply:
Is it important for this to be Arena-allocated? Seems to me like `compute_and_cache` will only be computed once per `VLoopPointers` instance, right?

Reply:
> Will the pointer ever change? Could potentially change this to a reference.
I could make it a reference. But data structures like `GrowableArray` take a `Arena*`. So then I have to use `*` and `&` all the time. I don't like that, it makes the code much more "noisy".